Restore icons page (#2889)
Orange-OpenSource/Orange-Boosted-Bootstrap: site/content/docs/0.3/extend/icons.md

## Icons rendering

Icons are designed within a square layout to preserve consistency. Within this square, there exists a designated safety zone to guarantee that icons can be used in various sizes and contexts while maintaining alignment as intended by the designers. The dimensions of the icons encompass this safety zone, ensuring adaptability and consistency across diverse applications.

<svg width="8em" height="8em" class="bg-body-secondary" aria-hidden="true">
  <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#info"/>
</svg>
<svg width="8em" height="8em" class="bg-body-secondary" aria-hidden="true">
  <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#download"/>
</svg>
<svg width="8em" height="8em" class="bg-body-secondary" aria-hidden="true">
  <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#tick"/>
</svg>

## Icons color

To color icons, you can:
- use [color utilities classes]({{< docsref "utilities/colors#colors" >}}) like `.text-brand-primary`, `.text-disabled` or `.text-status-positive` that change current color value.

> - specify a color using CSS variables like `--bs-color-content-brand-primary`, `--bs-color-content-disabled"` or `--bs-color-content-status-positive"`.

Reviewer: ```suggestion
- specify a color using CSS variables like `--bs-color-content-brand-primary`, `--bs-color-content-disabled` or `--bs-color-content-status-positive`.
```

> For SVGs (SVG sprites, inline SVGs or CSS mask image SVGs), the `fill="currentColor"` attribute is required on the `<svg>` or `<symbol>` tag, if you want to change the icons color on the fly: it will make the icon inherit color from itself or its parent containers. It is set on the icons downloaded from Solaris icons finder. For CSS mask image SVGs, the `background-color="currentColor"` CSS property should be set.

Reviewer: ```suggestion
For SVGs (SVG sprites, inline SVGs or CSS mask image SVGs), the `fill="currentColor"` attribute is required on the `<svg>` or `<symbol>` tag, if you want to change the icons color on the fly: it will make the icon inherit color from itself or its parent containers. It is set on the icons downloaded from Solaris icons finder. For CSS mask image SVGs, the `background-color: currentColor` CSS property should be set.
```

  </svg>
  <span class="icon si si-settings text-muted decorative-sm-icon" aria-hidden="true"></span>
</p>
<p class="p-2 bg-emphasized d-inline-flex align-items-center gap-shorter" data-bs-theme="dark">
  <svg class="decorative-sm-icon" aria-hidden="true">
    <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#heart-recommend"/>
  </svg>
  <svg class="decorative-sm-icon" style="color: var(--bs-color-content-default)" aria-hidden="true">
    <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#delete"/>
  </svg>
  <svg class="decorative-sm-icon" style="color: var(--bs-color-content-brand-primary)" aria-hidden="true">
    <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#download"/>
  </svg>
  <svg class="decorative-sm-icon" style="color: var(--bs-color-content-disabled)" aria-hidden="true">
    <use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#trash"/>
  </svg>
  <span class="icon si si-settings decorative-sm-icon" style="color: var(--bs-color-content-muted)" aria-hidden="true"></span>
</p>
{{< /example >}}

## Icons size

> Icons' size depend on where they are placed in the page content, so please make sure to either use our utilities provided in this page or follow the design. Icons size utility classes are done to be used upon `<img>`, `<svg>` or even icon font wherever the icon is.

Reviewer: ```suggestion
Icons' size depends on where they are placed in the page content, so please make sure to either use our utilities provided in this page or follow the design. Icons size utility classes are designed to be used upon `<img>`, `<svg>` or even icon font, wherever the icon is.
```

{{< /callout >}}-->

### Inline SVG

This technique should only be used if you have few icons to render, and if they are used only once in your website.

You can embed your icons directly within the HTML of your page (as opposed to an external image file). This way to use SVGs can benefit of the power of `currentColor` for easy coloring and theming.

{{< example class="mt-0" >}}
<svg fill="currentColor" class="decorative-sm-icon" viewBox="0 0 1000 1000" aria-hidden="true">
  <path d="M656.7 422.409a229.96 229.96 0 0 1-315.39.008A224.95 224.95 0 0 0 224.064 615H224v210a100 100 0 0 0 100 100h450V620a224.94 224.94 0 0 0-117.3-197.591M679 255A180 180 0 1 1 499 75a180 180 0 0 1 180 180" style="fill-rule:evenodd"/>
</svg>
{{< /example >}}

### SVG external image

You can use SVG through an `<img>` tag when:
- you only have few icons to render
- you don't need to change the icons colors. For that reason, **we do not recommend to use that technique**, so it won't be documented here.

### CSS mask image SVG

CSS mask image icons should be used when you can't or don't want to refer to an external image, or if you want to include the icon in a CSS `::before` or `::after` pseudo-element.

> You should use the `mask-image` property instead of `background-image` to benefit from the coloring and theming system, by setting `background-color` property to `currentColor`. You can use the SVG code within your CSS (be sure to escape any characters with [our internal `escape-svg()` function]({{< docsref "/customize/sass" >}}#escape-svg)). When no dimensions are specified via `width` and `height` on the `<svg>`, the icon will fill the available space. Mask images can alse be styled through CSS properties like `width` or `height`. Note that the `xmlns` attribute is required.

Reviewer 2: ```suggestion
You should use the `mask-image` property instead of `background-image` to benefit from the coloring and theming system, by setting `background-color` property to `currentColor`. You can use the SVG code within your CSS (be sure to escape any characters with [our internal `escape-svg()` function]({{< docsref "/customize/sass" >}}#escape-svg)). When no dimensions are specified via `width` and `height` on the `<svg>`, the icon will fill the available space. Mask images can also be styled through CSS properties like `width` or `height`. Note that the `xmlns` attribute is required.
```
